flink-table/flink-table-planner/src/test/resources/restore-tests/stream-exec-calc_1/async-calc-condition/plan/async-calc-condition.json

```diff
@@ -0,0 +1,186 @@
+{
+  "flinkVersion" : "1.19",
+  "nodes" : [ {
+    "id" : 20,
+    "type" : "stream-exec-table-source-scan_1",
+    "scanTableSource" : {
+      "table" : {
+        "identifier" : "`default_catalog`.`default_database`.`source_t`",
+        "resolvedTable" : {
+          "schema" : {
+            "columns" : [ {
+              "name" : "a",
+              "dataType" : "INT"
+            } ],
+            "watermarkSpecs" : [ ]
+          },
+          "partitionKeys" : [ ]
+        }
+      },
+      "abilities" : [ {
+        "type" : "FilterPushDown",
+        "predicates" : [ ]
+      } ]
+    },
+    "outputType" : "ROW<`a` INT>",
+    "description" : "TableSourceScan(table=[[default_catalog, default_database, source_t, filter=[]]], fields=[a])",
+    "inputProperties" : [ ]
+  }, {
+    "id" : 21,
+    "type" : "stream-exec-calc_1",
+    "projection" : [ {
+      "kind" : "INPUT_REF",
+      "inputIndex" : 0,
+      "type" : "INT"
+    }, {
+      "kind" : "CALL",
+      "syntax" : "SPECIAL",
+      "internalName" : "$CAST$1",
+      "operands" : [ {
+        "kind" : "INPUT_REF",
+        "inputIndex" : 0,
+        "type" : "INT"
+      } ],
+      "type" : "BIGINT"
+    } ],
+    "condition" : null,
+    "inputProperties" : [ {
+      "requiredDistribution" : {
+        "type" : "UNKNOWN"
+      },
+      "damBehavior" : "PIPELINED",
+      "priority" : 0
+    } ],
+    "outputType" : "ROW<`a` INT, `f0` BIGINT>",
+    "description" : "Calc(select=[a, CAST(a AS BIGINT) AS f0])"
+  }, {
+    "id" : 22,
+    "type" : "stream-exec-async-calc_1",
+    "configuration" : {
+      "table.exec.async-scalar.buffer-capacity" : "10",
```

Comment:
It would be nice to have at least one test that actually tests the parameters. Not sure how feasible this is but would it be possible to come up with a function that fails for 2 attempts and the the 3rd attempt actually happens after restore (by matching against a certain input arg for example?).

Reply:
Ok, I was able to do this.  I fail on the input 2 which is produced before restore.  Then after the restore occurs, I succeed 2 after number of retries, and check for 2 in the output post-restore.

I considered some other configs to test and most can't really be adjusted without failing the job.  For example, timeout is total time (not time before retry), so I can't let it timeout and then continue post-restore.  Same with NO_RETRY, since that would fail the job immediately.